tests/test_log_likelihood.py

```diff
@@ -0,0 +1,88 @@
+"""Assert values log-likelihood function under ambiguity (speed-up)"""
+import os
+from pathlib import Path
+
+import numpy as np
+import pytest
+import respy as rp
+from respy.likelihood import get_crit_func
+from respy.simulate import get_simulate_func
+
+subdir = Path(f"{os.environ['PROJECT_ROOT']}/tests/resources")
+
+
+# Test: Assert likelihood value for different model specifications (options, params)
+# Includes ambiguity for iteration i > len(model_spec_params)/2
+# Reference data was generated with inclusion of ambiguity but w/o speedup
+def test_loglike_around_models():
+    # Load model specifications and associated loglike values
+    model_spec_params = np.load(
+        f"{subdir}/model_spec_params.npy", allow_pickle="TRUE"
+    ).item()
+    model_spec_options = np.load(
+        f"{subdir}/model_spec_options.npy", allow_pickle="TRUE"
+    ).item()
+    model_spec_loglike_values = np.load(
+        f"{subdir}/loglike_values.npy", allow_pickle="TRUE"
+    ).item()
+
+    for i in range(len(model_spec_params)):
+        # Initialize values of current iteration
+        params = model_spec_params[str(i)]
```

Comment:
Why is the conversion to string needed? Can't we just set it up as floats during the generation of the test battery?